app/helpers/theme_helper.rb

```diff
@@ -70,40 +61,26 @@ def theme_color
   end
 
   def mobile_theme_color
-    theme = get_active_theme
+    theme = active_theme_user&.theme
     if theme
       theme.mobile_theme_color
     else
       "#f0edf4"
     end
   end
 
-  def get_active_theme
-    if @user&.theme
-      if user_signed_in?
-        if current_user&.show_foreign_themes?
-          @user.theme
-        else
-          current_user&.theme
-        end
-      else
-        @user.theme
-      end
-    elsif @answer&.user&.theme
-      if user_signed_in?
-        if current_user&.show_foreign_themes?
-          @answer.user.theme
-        else
-          current_user&.theme
-        end
-      else
-        @answer.user.theme
-      end
-    elsif current_user&.theme
-      current_user.theme
+  def active_theme_user
+    user = @user ||= @answer&.user
```

Comment:
would avoid assigning to `@user`, who knows what this can break
```suggestion
    user = @user || @answer&.user
```